### DISCUSSION TITLE
Footsteps length would transform OpenTracks in a pedometer, too

### DISCUSSION THREAD

**Discussion by chipechop** [OpenTracksApp/OpenTracks#751] [Ideas]
A bunch of new features could be added, if "Footstep length" was added, aside to "Wheel size": by dividing the distance covered by footstep length, we would know the **footstep count** of a walk!

Opentracks would be a pedometer.

You could suggest to the user to walk ten footsteps at home and then measure the distance from the heel of the first footstep, to the heel of the last: the measure of a single footstep would then be 1/10 of the total. 

The measurements of the footsteps that a user would have to save could be of two kinds: relaxed walk (4.5 km/hr), fast walk (7.0 km/hr). So there would be three measurements: wheel circumference, relaxed footsteps, fast footsteps.

Once the footsteps length was known: you could count them with a simple division.

**Comment by dennisguse** [comment]
@chipechop Thanks for reaching out!

Actually, most Android phones have a pedometer (virtual sensor based on accelerometer + gyroscope), so no need to compute it.
Sadly, OpenTracks does not use this sensor, but we have an issue for this: https://github.com/OpenTracksApp/OpenTracks/issues/463

Regarding cycling: here we get wheel rotation count from a sensor and use this compute the distance/speed as it is better than anything provided by GPS.

**Reply by chipechop** [reply]
I don't know pedometers that respect your privacy, as Open Tracks do.

The "pedometer" I'm thinking about wouldn't use sensors: it would just divide the distance of the track, by the step length manually set into the preferences of Open Tracks, to give an approximate count of the steps.

I could do the math by myself (7km/70cm=10.000 footsteps), but adding it as a rough estimate of the steps of a track would be more than welcome.

The point, for me, is privacy: don't want to share anything with anybody, to measure my steps.

**Reply by dennisguse** [reply]
My point was that your phone may actually be able to count the steps.
What is the benefit of deriving them from the distance.
Especially as the length is different depending on the terrain, tempo etc.
